sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -10,3 +10,175 @@
 
 #include <_az_cfg.h>
 
+static const az_span provisioning_service_api_version
+    = AZ_SPAN_LITERAL_FROM_STR("/api-version=" AZ_IOT_PROVISIONING_SERVICE_VERSION);
+
+static const az_span str_dps = AZ_SPAN_LITERAL_FROM_STR("$dps");
+static const az_span str_registrations = AZ_SPAN_LITERAL_FROM_STR("/registrations/");
+static const az_span str_put_iotdps_register
+    = AZ_SPAN_LITERAL_FROM_STR("PUT/iotdps-register/?$rid=1");
```

Comment:
request id number doesn't change?

Reply:
No: I don't see any value in doing that: RID is fixed. Retry must disconnect.